src/algebra/module/pointwise_pi.lean

```diff
@@ -0,0 +1,50 @@
+/-
+Copyright (c) 2021 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import algebra.pointwise
+import algebra.module.pi
+
+/-!
+# Pointwise actions on sets in Pi types
+
+This file contains lemmas about pointwise actions on sets in Pi types.
+
+## Tags
+
+set multiplication, set addition, pointwise addition, pointwise multiplication, pi
+
+-/
+
+open_locale pointwise
+open set
+
+variables {K ι : Type*} {R : ι → Type*}
+
+@[to_additive]
+lemma smul_pi_subset [∀ i, has_scalar K (R i)] (r : K) (t : Π i, set (R i)) (s : set ι) :
+  r • pi s t ⊆ pi s (r • t) :=
+begin
+  rintros x ⟨y, h, rfl⟩ i hi,
+  exact smul_mem_smul_set (h i hi),
+end
+
+@[to_additive]
+lemma smul_univ_pi [∀ i, has_scalar K (R i)] (r : K) (t : Π i, set (R i)) :
+  r • pi (univ : set ι) t = pi (univ : set ι) (r • t) :=
+subset.antisymm (smul_pi_subset _ _ _) $ λ x h, begin
+  refine ⟨λ i, classical.some (h i $ set.mem_univ _), λ i hi, _, funext $ λ i, _⟩,
+  { exact (classical.some_spec (h i _)).left, },
+  { exact (classical.some_spec (h i _)).right, },
+end
+
+@[to_additive]
+lemma smul_pi [group K] [∀ i, mul_action K (R i)] (r : K) (t : Π i, set (R i)) (S : set ι) :
```

Comment:
One final nit: I'd normally put the arguments in the order they appear in the lemma, so
```suggestion
lemma smul_pi [group K] [∀ i, mul_action K (R i)] (r : K) (S : set ι) (t : Π i, set (R i)) :
```
same for the lemmas below